sycl/source/detail/event_impl.cpp

```diff
@@ -250,16 +250,25 @@ void event_impl::cleanupCommand(
     detail::Scheduler::getInstance().cleanupFinishedCommands(std::move(Self));
 }
 
+void event_impl::checkProfilingPreconditions() const {
+  if (MQueue.lock() != nullptr) {
+    if (!MQueue.lock()->has_property<property::queue::enable_profiling>()) {
```

Comment:
```suggestion
  QueueImplPtr QueueImpl = MQueue.lock();
  if (QueueImpl && !QueueImpl->has_property<property::queue::enable_profiling>()) {
```
It is preferable to only create one shared pointer to minimize the reference-counting overhead.

Reply:
Ok, I'll change it.